src/java/org/apache/cassandra/index/sai/plan/StorageAttachedIndexQueryPlan.java

```diff
@@ -94,7 +94,7 @@ public static StorageAttachedIndexQueryPlan create(ColumnFamilyStore cfs,
     @Override
     public Set<Index> getIndexes()
     {
-        return indexes;
+        return ImmutableSet.copyOf(indexes);
```

Comment:
I'm not sure I understand the defensive copy when all callers do w/ `indexes` is read. If the worry is that `indexes` will change underneath us, it seems like we'd want to copy/snapshot when `StorageAttachedIndexGroup` creates the plan?

Reply:
This is pretty dumb for many reasons. Firstly, `create` already creates an `ImmutableSet` so there is no need to make it immutable again. Secondly (and the main reason), there is a casting issue because we are storing a `Set<StorageAttachedIndex>` rather than a `Set<Index>` even though the only use of the set is this method. I've changed this to class to create and use a `Set<Index>` and removed this copy.